sample-app/src/main/java/com/sinch/sample/numbers/NumbersSampleFlow.java

```diff
@@ -0,0 +1,392 @@
+
+    // 4. Check if 'US' is part of the regions that support the 'LOCAL' type
+    if (availableRegionListResponse.stream()
+        .noneMatch(region -> regionCode.equals(region.getRegionCode()))) {
+      throw new ApiException("The '" + regionCode + "' region is not supported.");
+    }
+
+    echo(
+        "The region '"
+            + regionCode
+            + "' is available. Let's use it in the following of this program.\n");
+
+    return regionCode;
+  }
+
+  String getAvailablePhoneNumber(
+      int step, SinchClient sinchClient, String regionCode, String codePattern) {
+
+    echoStep(
+        step,
+        "List available numbers in the '"
+            + regionCode
+            + "' region with pattern '"
+            + codePattern
+            + "'");
+
+    // 1. Build the request data for listing the numbers of type 'LOCAL' in the regionCode with an
+    // area code pattern
+    var availableNumbersRequestData =
+        AvailableNumberListAllRequestParameters.builder()
+            .setRegionCode(regionCode)
+            .setType(NumberType.LOCAL)
+            .setNumberPattern(
+                NumberPattern.builder()
+                    .setSearchPattern(SearchPattern.START)
+                    .setPattern(codePattern)
+                    .build())
+            .build();
+
+    // 2. Request for available numbers using the built-in SDK method
+    // This API is following the SDK pattern for list and the server response is wrapped inside a
+    AvailableNumberListResponse availableNumbersListResponse =
+        sinchClient.numbers().available().list(availableNumbersRequestData);
+
+    // 3. Looking for first available number
+    // auto iteration other items is available for all List responses.
+    // In case of underlying API is requiring new calls (paginated results): it will be handled
+    // automatically
+    AvailableNumber availableNumber =
+        availableNumbersListResponse.stream()
+            .findFirst()
+            .orElseThrow(
+                () ->
+                    new ApiException(
+                        "No numbers are available for the number type "
+                            + availableNumbersRequestData.getType()
+                            + " in the region "
+                            + availableNumbersRequestData.getRegionCode()
+                            + " with the pattern "
+                            + availableNumbersRequestData.getNumberPattern().get().getPattern()));
+
+    // 4. return the phone number
+    String phoneNumber = availableNumber.getPhoneNumber();
+    echo("The phone number '" + phoneNumber + "' is available. Let's rent it!\n");
+
+    return phoneNumber;
+  }
+
+  void rentPhoneNumber(int step, SinchClient sinchClient, String phoneNumber) {
+
+    echoStep(step, "Rent phone number: '" + phoneNumber + "'");
+
+    // 1. Build the request data
+    var rentRequestData = AvailableNumberRentRequestParameters.builder().build();
```

Comment:
Here we are not mentioning any sms or voice configuration. They are optional in the OAS documentation, but maybe it doesn't make sense to rent a number without any of these configuration (unless we update the active number accordingly in a subsequent API call)

Reply:
Right; ideally the `sinchClient.numbers().available().rent` should accept a `null` parameter or provide a second function without the `parameters` parameter.

I modified the `rent` function to accept null parameter but changed the sample too, to set a callback URL parameter: this is here the purpose if the sample: an on-boarding experience